draft-ietf-keytrans-protocol.md

```diff
@@ -1190,23 +1441,54 @@ struct {
 In turn, the Transparency Log responds with a SearchResponse structure:
 
 ~~~ tls-presentation
+struct {
+  opaque proof<VRF.Np>;
+  opaque commitment<Hash.Nh>;
+} BinaryLadderStep;
+
 struct {
   FullTreeHead full_tree_head;
 
-  SearchProof search;
-  opaque opening<16>;
+  optional<uint32> version;
+  BinaryLadderStep binary_ladder<0..2^8-1>;
+  CombinedTreeProof search;
+  InclusionProof inclusion;
+
+  opaque opening<Nc>;
   UpdateValue value;
 } SearchResponse;
 ~~~
 
+Each `BinaryLadderStep` structure contains information related to one version of
+the label that's in the binary ladder. The `proof` field contains the VRF
+proof, and `commitment` contains the commitment to the label's value at that
+version. The `binary_ladder` field contains these structures in the same order
+that the versions are output by the algorithm in {{binary-ladder}}.
+
+The `search` field contains the output of updating the user's view of the tree
+to match `FullTreeHead.tree_head.size` followed by either a fixed-version or
+greatest-version search for the requested label, depending on whether `version`
+was provided in `SearchRequest` or not. If searching for the greatest version of
+the label, this version is provided in `SearchResponse.version`; otherwise, the
+field is empty.
+
+The `inclusion` field contains a proof of inclusion for all of the log tree
+leaves where either a search proof was provided in
+`CombinedTreeProof.prefix_proofs` or the prefix tree root hash was provided
+directly in `CombinedTreeProof.prefix_roots`. If the user advertised a
+previously-observed tree size in `last`, the proof in `inclusion` also functions
+as a proof of consistency.
+
 Users verify a search response by following these steps:
 
-1. Evaluate the search proof in `search` according to the steps in
-   {{proof-combined-tree}}. This will produce a verdict as to whether the search
-   was executed correctly and also a candidate root value for the tree. If it's
-   determined that the search was executed incorrectly, abort with an error.
-2. With the candidate root value for the tree, verify the given `FullTreeHead`.
-3. Verify that the commitment in the terminal search step opens to
+1. Compute the VRF output for each version of the label from the proofs in
+   `binary_ladder`.
+2. Verify the proof in `search` as described in {{proof-combined-tree}}.
+3. Compute a candidate root value for the tree from the proof in `inclusion`,
+   the hashes of log entries used in `search`, and any previously-retained full
+   subtrees of the log tree.
+4. With the candidate root value for the tree, verify `FullTreeHead`.
```

Comment:
What does "verify" mean here? I think: check for equality of the hash, no? I think that's short enough s.t. it can be made explicit.

Reply:
Will specify this more clearly in a follow-up PR